### PR TITLE
Adding Delete Flag

### LINKED ISSUE #49: Create `--delete` Flag
It might be useful to have a `-d`/`--delete` flag that will do a destructive sync (delete any parameters that are not present in the JSON being used).

@cmac1000 @joshfinnie what do you guys think about this? Worth it?

### PR DESCRIPTION
**Pull request by yhakbar**
This adds a `--delete` flag that will delete the delta between the parameters being uploaded by a put, and the parameters that are not listed in the json that already exist in SSM.

In addition, a prompt is added to confirm this decision. This prompt can be overridden with the flag `--auto-approve`, and colors present in the prompt can be removed through the flag `--no-color` for accessibility purposes.

closes #49

### CODE REVIEW

File: .tool-versions [hunk 1]
```diff
@@ -0,0 +1 @@
+golang 1.14.5
```
Comment (joshfinnie) [inline .tool-versions:1]: tests are already being run on 1.15... is there a reason we chose 1.14.5 here?

Comment (yhakbar) [reply .tool-versions:1]: I think that was just a mistake. Not sure why I did that.

File: README.md [hunk 2]
```diff
@@ -10,11 +10,11 @@ This is an experimental library, and is currently unsupported.
...
 
-```
-$ gorson get /a/parameter/store/path/ > ./example.json
+```bash
+gorson get /a/parameter/store/path/ > ./example.json
 ```
 
-```
+```bash
 $ cat ./example.json
```
[the remark this answers is not in the document]
Comment (yhakbar) [reply README.md:18]: According to [this](https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md#md014---dollar-signs-used-before-commands-without-showing-output), it's standard practice to only include `$` when displaying output of a command.

File: README.md [hunk 3]
```diff
@@ -65,46 +65,95 @@ beta=the_beta_value
...
+gorson put /a/parameter/store/path/ --file=./new-values.json
 ```
-$ gorson put /a/parameter/store/path/ --file=./new-values.json
+
+## Delete parameter difference on put
+
+```bash
+$ gorson put /a/parameter/store/path/ --file=./different-values.json --delete
+
+The following are present in the file, but not in parameter store:
```
[the remark this answers is not in the document]
Comment (yhakbar) [reply README.md]: Good catch! Addressed [here](https://github.com/pbs/gorson/pull/51/commits/67755f970c8f6e007c3993c5e8a87e04bce34def).

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (joshfinnie) [reply README.md:18]: huh, that's super interesting!

**Review by joshfinnie — APPROVED** [review]

**Pull request merged**